### PR TITLE
Setup CI pipeline with Github Actions to deploy on Heroku

### PR DESCRIPTION
**Pull request by espoirMur**
#### Description of changes made

> This PR set up a CI pipeline with GitHub actions and deploy on Heroku
> **his PR will add** :

- Setup Github actions
- Deploy application on heroku

#### Type of change

> **Please select the relevant options**
- [x] Changes that affect the build system or external dependencies
- [ ] New feature (non-breaking changes which add the new functionalities)
- [ ] Bug fix (non-breaking changes which fix an issue)
- [ ] Documentation (added or improved the documentation)
- [ ] Breaking change (fix or a feature that would cause existing feature not to work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

> **Please describe the tests that you ran to verify your changes**

- After cloning the [repo](https://github.com/kisanola/AlphaCodeFrontend.git)
- Check out this branch by `branch here`
- Run `yarn or npm install` to install the project dependencies
- Run `ACTION TO BE EXECUTED`

- [ ] Unit
- [x] Integration
- [ ] End to end

#### How should this be manually tested?

> **Provide the steps to manually test the functionalities implemented in this PR**

#### Add the link to the to gitKraken board?
[Setup unit testing and continuous integration + push hook](https://app.gitkraken.com/glo/card/f966a62d8cba4dd191e2de7575175618)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have ensured coding conventions for my code prior to submission
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests which prove the effectiveness of my work
- [x] New and existing tests pass locally with my changes

#### PS :

- The remaining part is setting up the  MongoDB on heruku...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,10 +9,28 @@
     "dev-start": "ts-node ./src/index.ts",
     "dev": "nodemon -e ts -w ./src -x npm run dev-start",
     "build-dev": "tsc && npm start",
-    "build": "rm -rf dist && tsc cp -r ./src/templates ./dist/templates",
-    "lint": "tsc --noEmit && eslint \"src/*/**/*.{js,ts,tsx}\" --quiet --fix",
+    "build": "rm -rf dist && tsc",
+    "lint": "tsc --noEmit && eslint 'src/*/**/*.{js,ts,tsx}' --quiet --fix",
     "migrate": "ts-jest config:migrate jest.config.js"
   },
+  "jest": {
```
Comment (oesukam) [inline package.json:16]: What about putting these configurations`jest.config.js`?

**Review by oesukam — CHANGES_REQUESTED** [review]
Can you add the heroku's URL in the Readme an try to squash your commits to have a maximum of 3 in order to easier rebasing?

File: package.json (hunk 1 above)
Comment (espoirMur) [reply package.json:16]: Yeah, definitely , I will do it...

**Comment by oesukam** [comment]
@espoirMur Any updates?

**Comment by espoirMur** [comment]
> @espoirMur Any updates?

I am checking on this tonight...

**Comment by espoirMur** [comment]
> @espoirMur Any updates?

@oesukam the ci is done
- The remaining part is setting up the  MongoDB on heruku...

**Pull request merged**